### ISSUE TITLE
Should block ability to stumble into infinite while loop in percentile calculation for dff

### ISSUE THREAD

**Issue by EricThomson** [flatironinstitute/CaImAn#1262]
When calculating dff there is a percentile calculation that has a try/except block in a while loop. If the exception is triggered every run, the while loop never breaks.  😱  I ran into this today. We should fix this design flaw

https://github.com/flatironinstitute/CaImAn/blob/344a7e5651736d99f77c0be90928d517fda0b9e9/caiman/utils/stats.py#L184

**Comment by EricThomson** [comment]
Yikes this is a consequence of functions being deprecated in scipy. Will temporarily lock to < 1.12 for now and am adding to roadmap to fix.

**Comment by kushalkolar** [comment]
Oh my, I cannot comprehend the amount of :spaghetti: in that function. This is also an example of why bare `except` is generally a bad idea. From a first glance this could be remedied by adding a `finally` block and putting a counter inside the `except` to force exiting the loop if more than `n` exceptions are raised. But given how deeply nested that function is, I think it can just be designed better but I don't have time for that.

**Comment by EricThomson** [comment]
Kushal -- I started with a counter, but I ended up unhappy with this approach as it leaves the smelly code in there. 😄 As you said, it can be designed better. The problem in this case is that it depends (after you go down the stack enough) on a deprecated function, so it will always hit the counter limit. I need to do a real sweep of Caiman for soon-to-be deprecated functions! There are lots of them.
